doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+   usage.
+ * A custom class must implement three functions:
+
+```c++
+// You can use this as a starting point for implementation.
+class CustomFitnessFunction
+{
+  // Compute the gain for the given vector of values, where `values[i]` has an
+  // associated instance weight `weights[i]`.
+  //
+  // `RowType` and `WeightVecType` will be vector types following the Armadillo
+  // API.  If `UseWeights` is `false`, then the `weights` vector should be
+  // ignored (e.g. the responses are not weighted).
+  //
+  // In the version with `begin` and `end` parameters, only the subset between
+  // `labels[begin]` and `labels[end]` (inclusive) should be considered.
+  template<bool UseWeights, typename RowType, typename WeightVecType>
+  double Evaluate(const RowType& labels,
+                  const WeightVecType& weights);
+
+  template<bool UseWeights, typename RowType, typename WeightVecType>
+  double Evaluate(const RowType& labels,
+                  const WeightVecType& weights,
+                  const size_t begin,
+                  const size_t end);
+
+  // Return the output value for prediction for a leaf node whose training
+  // values are made up of the values in the vector `responses` (optionally with
+  // associated instance weights `weights`).
+  //
+  // `ResponsesType` and `WeightsType` will be vector types following the
+  // Armadillo API.  If `UseWeights` is `false`, then the `weights` vector
+  // should be ignored (e.g. the responses are not weighted).
+  template<bool UseWeights, typename ResponsesType, typename WeightsType>
+  double OutputLeafValue(const ResponsesType& responses,
+                         const WeightsType& weights);
+};
+```
+
+***Note:*** this API differs from the `FitnessFunction` API required for
+[`DecisionTree`](decision_tree.md)! <!-- TODO: fix link! -->
+
+---
+
+#### `DimensionSelectionType`
+
+ * When splitting a decision tree, `DimensionSelectionType` proposes possible
+   dimensions to try splitting on.
+ * `MultipleRandomDimensionSelect` _(default)_ is available for drop-in usage
+   and proposes a different random subset of dimensions at each decision tree
+   node.
+    - By default each random subset is of size `sqrt(d)` where `d` is the number
+      of dimensions in the data.
+    - If constructed as `MultipleRandomDimensionSelect(n)` and passed to the
+      constructor of `RandomForestRegressor` or the `Train()` function, each 
+      random subset will be of size `n`.
+ * Each `RandomForestRegressor` [constructor](#constructors) and each version of
+   the [`Train()`](#training) function optionally accept an instantiated
+   `DimensionSelectionType` object as the very last parameter (after
+   `maxDepth` in the constructor, or `warmStart` in `Train()`), in case some
+   internal state in the dimension selection mechanism is required.
+ * A custom class must implement three simple functions:
+
+
+```c++
+class CustomDimensionSelect
```

Comment:
Just a sanity check here, I am assuming that this is the exact same API required by the regular `RandomForest`.